x-pack/plugins/canvas/public/components/sidebar_header/custom_element_modal.tsx

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/* eslint-disable react/forbid-elements */
+import React, { PureComponent } from 'react';
+import { get } from 'lodash';
+import PropTypes from 'prop-types';
+import {
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiOverlayMask,
+  EuiFormRow,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiSpacer,
+  EuiTextArea,
+  EuiText,
+  EuiTitle,
+  EuiModalHeaderTitle,
+  EuiModalHeader,
+  EuiIcon,
+  // @ts-ignore EuiCard is an exported member of @elastic/eui
+  EuiCard,
+  // @ts-ignore EuiFilePicker is an exported member of @elastic/eui
+  EuiFilePicker,
+} from '@elastic/eui';
+import { VALID_IMAGE_TYPES } from '../../../common/lib/constants';
+import { encode } from '../../../common/lib/dataurl';
+
+const MAX_NAME_LENGTH = 20;
+const MAX_DESCRIPTION_LENGTH = 100;
+
+export interface Props {
+  /**
+   * name of the custom element
+   */
+  name?: string;
+  /**
+   * description of the custom element
+   */
+  description?: string;
+  /**
+   * preview image of the custom element as a base64 dataurl
+   */
+  image?: string;
+  /**
+   * title of the modal
+   */
+  title: string;
+  /**
+   * A click handler for the save button
+   */
+  onSave: (name: string, description?: string, image?: string) => void;
+  /**
+   * A click handler for the cancel button
+   */
+  onCancel: () => void;
+}
+
+export class CustomElementModal extends PureComponent<Props> {
+  public static propTypes = {
+    name: PropTypes.string,
+    description: PropTypes.string,
+    image: PropTypes.string,
+    title: PropTypes.string.isRequired,
+    onSave: PropTypes.func.isRequired,
+    onCancel: PropTypes.func.isRequired,
+  };
+
+  public state = {
+    name: this.props.name || '',
```

Comment:
You end up mixing the use of `null` and `''` below.  I'd default these to `null` if `EuiCard` can handle a `null` `title` property.  YMMV... but the null default on 163 stands out.

Reply:
It was only mixed for `image`. I switched `image` fallback value in state to `null` and removed the fallback value on line 163. I need `name` and `description` to be strings so I can get the length to update the remaining character help text, so I left them as empty strings.